compiler/rustc_lint/messages.ftl

```diff
@@ -420,6 +414,12 @@ lint_incomplete_include =
 
 lint_inner_macro_attribute_unstable = inner macro attributes are unstable
 
+lint_instantly_dangling = getting a pointer from a temporary `{$ty}` will result in a dangling pointer
```

Comment:
I would prefer if we would have a message that followed the lint name, i.e. signalling the dangling pointer first and then the temporary, as the issue is first and foremost the dangling pointer.

Maybe something like this?
```suggestion
lint_instantly_dangling = pointer to an immediately dropped temporary result in a dangling pointer
```